chpl/chpl-api/src/main/java/gov/healthit/chpl/registration/RateLimitingInterceptor.java

```diff
@@ -0,0 +1,122 @@
+package gov.healthit.chpl.registration;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+
+import javax.annotation.PreDestroy;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.EnvironmentAware;
+import org.springframework.context.MessageSource;
+import org.springframework.context.i18n.LocaleContextHolder;
+import org.springframework.context.support.DefaultMessageSourceResolvable;
+import org.springframework.core.env.Environment;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.handler.HandlerInterceptorAdapter;
+
+@Component
+public class RateLimitingInterceptor extends HandlerInterceptorAdapter implements EnvironmentAware{
+ 
+    private static final Logger logger = LogManager.getLogger(RateLimitingInterceptor.class);
+    
+    @Autowired
+    private Environment env;
+    
+    @Autowired
+    private MessageSource messageSource;
+    
+    private String timeUnit;
+    
+    private int limit;
+ 
+    private Map<String, SimpleRateLimiter> limiters = new ConcurrentHashMap<>();
+    
+    private List<String> whitelist = new ArrayList<String>();
+    
+    public RateLimitingInterceptor(){}
+    
+    public String getMessage(final String messageCode, final String input, final String input2) {
```

Comment:
This is getting ubiquitous enough I think it should be in some util class

Reply:
I already made a util class for this and it's my PR related to validation changes. If we're going to merge the PRs in order, Brian could bring in my code from that and use the util already written? Another issue with leaving PRs sitting in Dev!

Reply:
That sounds like a plan